### PR TITLE
Ticket 1307

### PR DESCRIPTION
**Pull request by tpltnt**
Hi there,

this is a quick stab at [ticket 1307](https://tahoe-lafs.org/trac/tahoe-lafs/ticket/1307) to add support for "HTTP OPTIONS" requests.

Cheers,
tpltnt

### CODE REVIEW

File: src/allmydata/web/common.py [hunk 1]
```diff
@@ -387,6 +387,22 @@ def renderHTTP(self, ctx):
...
+        """
+        Handle HTTP OPTIONS request by adding appropriate headers.
+
+        :param ctx: client transaction from which request is extracted
+        :returns: str (empty)
+        """
+        from allmydata import __version__
+        req = IRequest(ctx)
+        req.setHeader("server", "Tahoe-LAFS gateway v" + __version__)
+        methods = ', '.join([m[7:] for m in dir(self) if m.startswith('render_')])
```
Comment (meejah) [inline src/allmydata/web/common.py]: Do we do this other places? I kind of think maybe we *shouldn't* stick the exact version in there (e.g. just "Tahoe-LAFS gateway")

**Comment by meejah** [comment]
Looks good to me. Not sure if the appveyor failures are related to this; seems unlikely...

**Comment by tpltnt** [comment]
Is there anything left which is stalling a merge?

File: src/allmydata/web/common.py (hunk 1 above)
Comment (tpltnt) [reply src/allmydata/web/common.py]: Not that I am aware of. I removed it.

**Comment by tpltnt** [comment]
I removed the version string and synced with master.

**Pull request merged**